staging/src/k8s.io/client-go/generic/type.go

```diff
@@ -79,8 +79,9 @@ type alsoApplier[T objectWithMeta, C namedObject] struct {
 	client TypeClient[T]
 }
 
-// NewNamespaced constructs a namespaced client.
-func NewNamespaced[T objectWithMeta](
+// New constructs a client, namespaced or not, with no support for lists or apply.
+// Non-namespaced clients are constructed by passing an empty namespace ("").
+func New[T objectWithMeta](
```

Comment:
`NewTypeClient` ?

Reply:
To avoid polluting the `k8s.io/client-gen/generic` package, I went with `k8s.io/client-gen/gentype`, and `Client…` to avoid stuttering, with `NewClient…` constructors.